expr.go

```diff
@@ -29,22 +29,25 @@ type Expr struct {
 	// The expression that defined this expression's base value, if any.
 	Base *Expr `json:"base,omitempty"`
 
+	// Ranges for the object's keys, if this is an object expression.
+	KeyRanges map[string]Range `json:"keyRanges,omitempty"`
+
 	// The fields below act as a discriminated union. Only one must be non-nil at any given time. If all fields are nil,
 	// then this Expr is a null literal expression.
 
 	// The literal value, if this is a literal expression (nil, bool, json.Number, or string)
 	Literal any `json:"literal,omitempty"`
 
-	// The interpolations, if this is a a string interpolation expression
+	// The interpolations, if this is a a string interpolation expression.
 	Interpolate []Interpolation `json:"interpolate,omitempty"`
 
-	// The property accessors, if this is a a symbol expression
+	// The property accessors, if this is a a symbol expression.
 	Symbol []PropertyAccessor `json:"symbol,omitempty"`
 
-	// The access, if this is an access expression
+	// The access, if this is an access expression.
 	Access *AccessExpr `json:"access,omitempty"`
 
-	// The list elements, if this is a list expression
+	// The list elements, if this is a list expression.
 	List []Expr `json:"list,omitempty"`
```

Comment:
I thought we renamed `List` to `Array`?

Reply:
Still need to do so in the public API :(

Reply:
(I deliberately left this part out for the moment)